io/src/obj_io.cpp

```diff
@@ -146,16 +146,16 @@ int
 pcl::MTLReader::read (const std::string& obj_file_name,
                       const std::string& mtl_file_name)
 {
-  if (obj_file_name.empty() || !boost::filesystem::exists (obj_file_name))
+  if (obj_file_name.empty ())
   {
-    PCL_ERROR ("[pcl::MTLReader::read] Could not find file '%s'!\n",
-               obj_file_name.c_str ());
+    PCL_ERROR ("[pcl::MTLReader::read] No OBJ file name given!\n");
     return (-1);
   }
 
-  if (mtl_file_name.empty())
```

Comment:
Move to `pcl::MTLReader::read (const std::string& mtl_file_path)`

Reply:
But removing this check here would change the behaviour: if `mtl_file_name` is empty here, then `mtl_file_path` below would contain `obj_file_path.parent_path()` (and maybe a slash or backslash at the end), meaning `mtl_file_path` would not be empty. So as I see it, it is necessary to keep this check here.